docs/resource-validation.md

```diff
@@ -0,0 +1,137 @@
+# Gateway API Resource Validation
+
+This document describes how NGINX Kubernetes Gateway (NKG) validates Gateway API resources.
+
+## Overview
+
+There are several reasons why NKG validates Gateway API resources:
+
+- *Robustness*, to gracefully handle invalid resources.
+- *Security*, to prevent malicious input from propagating to the NGINX configuration.
+- *Correctness*, to conform to the Gateway API specification for handling invalid resources.
+
+Ultimately, the goal is to ensure that NGINX continues to handle traffic even if invalid Gateway API resources were
+created.
+
+A Gateway API resource (a new resource or an update for the existing one) is validated by the following steps, some of
+which are provided by the Gateway API, while others are done by NKG:
+
+1. OpenAPI schema validation by the Kubernetes API server.
+2. Webhook validation by the Gateway API webhook.
+3. Webhook validation by NKG.
+4. Validation by NKG.
+
+To confirm that a resource is valid and accepted by NKG, check the `Accepted` condition in the resource status,
+which must have the Status field set to `True`. For example, in a status of a valid HTTPRoute, if NKG accepts a
+parentRef, the status of that parentRef will look like this:
+```
+Status:
+  Parents:
+    Conditions:
+      Last Transition Time:  2023-03-30T23:18:00Z
+      Message:               The route is accepted
+      Observed Generation:   2
+      Reason:                Accepted
+      Status:                True
+      Type:                  Accepted
+    Controller Name:         k8s-gateway.nginx.org/nginx-gateway-controller
+    Parent Ref:
+      Group:         gateway.networking.k8s.io
+      Kind:          Gateway
+      Name:          gateway
+      Namespace:     default
+      Section Name:  http
+```
+
+> Make sure that the reported observed generation is the same as the resource generation.
+
+The remaining part of this document describes each step in detail with examples of how validation errors are reported.
```

Comment:
Passive voice. Maybe:
> The rest of this document describes each validation step in detail and gives examples of possible validation errors.

Reply:
I really want to retain the verb reported, because I'd like the readers to see examples how those errors are _reported_